lib/cosmos/script/scripting.rb

```diff
@@ -160,6 +160,29 @@ def check_raw(*args)
       _check(*args) {|tgt,pkt,item| tlm_raw(tgt,pkt,item) }
     end
 
+    # Executes the passed method and expects and exception to be raised.
+    # Raises a CheckError if an Exception is not raised.
+    # Usage:
+    #   check_exception { cmd(target_name, cmd_name, cmd_params = {}) }
+    # or
+    #   check_exception { cmd('target_name cmd_name with cmd_param1 value1, cmd_param2 value2') }
+    def check_exception(&block)
+      method = ''
+      begin
+        begin
+          # Try to get the block source. This doesn't always work ...
+          method = block.source.split('{')[1].split('}')[0].strip
+        rescue => error # Rescue errors with getting block source
+          method = block.source_location # source_location always works
```

Comment:
Because it only gives the source location and not the actual code. But since I can't figure out how block.source works and why it sometimes fails I think I'll have to just use source_location.